R/database.R

```diff
@@ -2628,3 +2630,385 @@ get_gene_expression = function(metadata,
+      }
+    }
+  }
+
+  #get metadata and subset based on the provided sample ID
+  if(!missing(this_sample_id)){
+    if(missing(these_samples_metadata)){
+      these_samples_metadata = get_gambl_metadata() %>%
+        dplyr::filter(sample_id == this_sample_id)
+      if(!nrow(these_samples_metadata==1)){
+        stop("Fetched metadata contains more than one row, or the data frame is empty (subset to the provided sample ID). Considder giving this funciton an already subset metadata with the sample ID of interest...")
+      }
+    }
+  }
+
+  #last check
+  if(missing(this_sample_id) && missing(these_samples_metadata)){
+    stop("please provide either a sample ID (this_sample_id) or an already subset metadata (these_samples_metadata)...")
+  }
+
+  #get wildcards
+  tumour_sample_id = this_sample_id
+  unix_group = pull(these_samples_metadata, unix_group)
+  seq_type = pull(these_samples_metadata, seq_type)
+  genome_build = projection
+  pairing_status = pull(these_samples_metadata, pairing_status)
+
+  if(pairing_status == "matched"){
+    normal_sample_id = pull(these_samples_metadata, normal_sample_id)
+  }else{
+    normal_sample_id = config::get("unmatched_normal_ids")[[unix_group]][[seq_type]][[genome_build]]
+  }
+
+  if(from_flatfile){
+    if(!remote_session){
+      path_template = config::get("results_flatfiles")$sv_manta$template
+      path_template = paste0(config::get("project_base"), path_template)
+      bedpe_path = glue::glue(path_template)
+
+      #if file does not exist, look for it in the merge (already loaded and given to the all_sv parameter, should be in the genome projection your interested in).
+      if(!file.exists(bedpe_path)){
+        if(missing(all_sv_grch37) && missing(all_sv_hg38)){
+          stop("Considder use all_sv parameter for retreiving SV calls for missing samples...")
+        }else{
+          if(projection == "grch37"){
+            bedpe_dat = all_sv_grch37 %>%
+              dplyr::filter(normal_sample_id == this_sample_id) #look for this_sample_id in the normal_sample_id column.
+            if(!nrow(bedpe_dat==0)){ #if sample not available, look for it in the tumour_sample_id.
+              bedpe_dat = all_sv_grch37 %>%
+                dplyr::filter(tumour_sample_id == this_sample_id)
+            }
+          }
+          if(projection == "hg38"){
+            bedpe_dat = all_sv_hg38 %>%
+              dplyr::filter(normal_sample_id == this_sample_id) #look for this_sample_id in the normal_sample_id column.
+            if(!nrow(bedpe_dat==0)){ #if sample not available, look for it in the tumour_sample_id.
+              bedpe_dat = all_sv_hg38 %>%
+                dplyr::filter(tumour_sample_id == this_sample_id)
+            }
+          }
+        }
+
+        #if can't find the sample in the existing merge, look for individual files (in the opposite projection) and lift over to the desired projection.
+        if(!nrow(bedpe_dat==0)){
```

Comment:
This code is assuming that if a file is missing then the genome build must be opposite to the projection. The actual genome_build is in the metadata. What if a file is missing for some other reason?

Reply:
Yes, I added this as a "last resort" type of step, if the sample is not available in the merge, or individual flat-file -> look for the sample (flat-file) in the opposite genome projection and lift it over to the desired projection. I can add another statement notifying the user if this requested file (flat file from the opposite projection) is not available (i.e missing for some other reason).

Reply:
is this resolved?